purchase_order_approval_block/models/purchase_order.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    approval_block_id = fields.Many2one(
+        comodel_name='purchase.approval.block.reason',
+        string='Approval Block Reason',
+    )
+    approval_blocked = fields.Boolean('Approval Blocked',
+                                      compute='_compute_approval_blocked')
+
+    @api.multi
+    def _compute_approval_blocked(self):
+        for rec in self:
+            if rec.approval_block_id:
+                rec.approval_blocked = True
+
+    @api.model
+    def create(self, vals):
```

Comment:
Instead of using write and create, use the `_track_subtype` function of mail.thread like in https://github.com/odoo/odoo/blob/8fbee7978f154bfd7d1424709cd66796cb3d82f3/addons/account/models/account_invoice.py#L1134

Reply:
Hi @rousseldenis,
This module is used as the base of #446. For this reason and from my point of view, it is necessary that it is in the create method that we send the notifications as in the other module this gets more restrictive (notifications are not send that frequent).

Reply:
I understand your purpose.

But in my point of view this is a little bit dirty to put such things in create and write.

In the track_subtype, why not using a method that can be surcharged to launch or not the notification ?

But it's my opinion :-)